### ISSUE TITLE
The `RegularGridInterpolator` requires that the points value be strictly **increasing**, not merely monotonic, not specified in docs and differs from scipy

### ISSUE THREAD

**Issue by anielsen001** [jax-ml/jax#36499]
### Description

The function can be found here:
https://github.com/jax-ml/jax/blob/753f089dad98728d6444f732323e020f109fec0d/jax/_src/third_party/scipy/interpolate.py#L36-L163

The docstring includes this example which works:
```python
points = (jnp.array([1, 2, 3]), jnp.array([4, 5, 6]))
values = jnp.array([[10, 20, 30], [40, 50, 60], [70, 80, 90]])
interpolate = RegularGridInterpolator(points, values, method='linear')
query_points = jnp.array([[1.5, 4.5], [2.2, 5.8]])
interpolate(query_points)
```
returns `Out[8]: Array([30., 64.], dtype=float32)`

However, simply reversing on of the axis directions results in all outputs as NaN:
```python
points = (jnp.array([1, 2, 3]), jnp.array([6, 5, 4]))
values = jnp.array([[10, 20, 30], [40, 50, 60], [70, 80, 90]])
interpolate = RegularGridInterpolator(points, values, method='linear')
query_points = jnp.array([[1.5, 4.5], [2.2, 5.8]])
interpolate(query_points)
```
returns `Out[19]: Array([nan, nan], dtype=float32)`

My workaround fix is to reorder things before creation of the jax objects so that the axis values are increasing.

This behavior differs from SciPy's implementation which only requires that the axis values be monotonic, so this doesn't serve as a direct drop in replacement without a fix. I think it can be addressed with some `jnp.flip` options internally. I'm happy to work on a PR if this is of interest. At the very least I would like to update the documentation to explain the difference.

### System info (python version, jaxlib version, accelerator, etc.)

In [25]: jax.print_environment_info()
jax:    0.9.2
jaxlib: 0.9.2
numpy:  2.4.2
python: 3.12.8 (main, Jan 11 2025, 09:42:09) [GCC 14.2.0]
device info: cpu-1, 1 local devices"
process_count: 1
platform: uname_result(system='Linux', node='none', release='6.17.0-14-generic', version='#14~24.04.1-Ubuntu SMP PREEMPT_DYNAMIC Thu Jan 15 15:52:10 UTC 2', machine='x86_64')

**Comment by kedar49** [comment]
@precision @mjwillson Assign me this , I'd love to work on this
